src/IO/H5.jl

```diff
@@ -4,28 +4,70 @@ macro namevar(x)
     name = split(string(x), ".")[end]
     return quote
         tmp = $(esc(x))
-        $(esc(name)), Array(tmp)
+        if tmp isa Float64
+            $(esc(name)), tmp
+        else
+            $(esc(name)), Array(tmp)
+        end
     end
 end
 
 """
-    checkpointing(dst, stokes, T, η, time)
+    checkpointing_hdf5(dst, stokes, T, η, time)
 
 Save necessary data in `dst` as and HDF5 file to restart the model from the state at `time`
 """
-function checkpointing(dst, stokes, T, η, time)
+function checkpointing_hdf5(dst, stokes, T, time)
     !isdir(dst) && mkpath(dst) # creat folder in case it does not exist
     fname = joinpath(dst, "checkpoint")
     h5open("$(fname).h5", "w") do file
         write(file, @namevar(time)...)
         write(file, @namevar(stokes.V.Vx)...)
         write(file, @namevar(stokes.V.Vy)...)
         write(file, @namevar(stokes.P)...)
+        write(file, @namevar(stokes.viscosity.η)...)
         write(file, @namevar(T)...)
```

Comment:
```suggestion
    h5open("$(fname).h5", "w") do file
        write(file, @namevar(time)...)
        write(file, @namevar(stokes.V.Vx)...)
        write(file, @namevar(stokes.V.Vy)...)
        write(file, @namevar(stokes.V.Vz)...)
        write(file, @namevar(stokes.P)...)
        write(file, @namevar(stokes.viscosity.η)...)
        write(file, @namevar(T)...)
```
Need also to save `stokes.V.Vz` for the 3D case. This arrays is `Nothing` in the 2D case so will take no space in the HD.